theano/tensor/basic.py

```diff
@@ -630,9 +630,14 @@ def get_scalar_constant_value(orig_v, elemwise=True,
                 v = v.owner.inputs[0]
                 continue
             elif isinstance(v.owner.op, theano.compile.ops.Shape_i):
-                if isinstance(v.owner.inputs[0], Constant):
-                    return numpy.asarray(
-                        v.owner.inputs[0].data.shape[v.owner.op.i])
+                i = v.owner.op.i
+                inp = v.owner.inputs[0]
+                if isinstance(inp, Constant):
+                    return numpy.asarray(inp.data.shape[i])
+                # The shape of a broadcastable dimension is 1
+                if inp.type.broadcastable[i]:
+                    return numpy.asarray(1)
```

Comment:
This is good. Why it was needed? @Sentient07 add a test for this in theano/tensor/test_basic.py in the class: T_get_scalar_constant_value